integration/src/main/java/scripts/testscripts/DatasetPermissionOperations.java

```diff
@@ -121,14 +121,12 @@ private void testDatasetReaderPermissions(
       throws ApiException {
     CreateDatasetRequest request = datasetRequest(storageSourceId, storageSystem);
     // note if this assertion fails we'll leave behind a stray dataset in the db
-    assertThrows(ApiException.class, () -> userDatasetsApi.createDataset(request));
-    assertThat(
-        userDatasetsApi.getApiClient().getStatusCode(),
-        is(HttpStatusCodes.STATUS_CODE_UNAUTHORIZED));
+    var exception = assertThrows(ApiException.class, () -> userDatasetsApi.createDataset(request));
```

Comment:
Why isn't this using the `createDatasetWithHttpInfo` that was used above? Is it because the response body includes the status code?

Reply:
It's because this throws an exception. A non-20x response code will cause an `ApiException` to be thrown, so it needs to be called inside `assertThrows()` or some other `try` form; even if we called the `WithHttpInfo()` version here it would still throw and never return a response. Once we have the exception we can look at the http code in the exception (which I think we would've done before as well).